### PR TITLE
Determine fixedip during nodeup directly

### PR DESCRIPTION
**Pull request by olemarkus**
As discussed in #9554 

This will break the task dependency cycle. It is a bit hackish with directly using protokube-code. That part should be moved to something shared, but perhaps best done in a followup PR?

### CODE REVIEW

**Comment by johngmyers** [comment]
First commit message has typo "defauls"

File: pkg/model/openstackmodel/servergroup.go [hunk 1]
```diff
@@ -164,6 +161,9 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 				}
 				c.AddTask(t)
 			case kops.InstanceGroupRoleMaster:
+				if b.Cluster.Spec.Topology != nil && b.Cluster.Spec.Topology.Masters == kops.TopologyPrivate {
+					continue
+				}
```
Comment (johngmyers) [inline pkg/model/openstackmodel/servergroup.go:166]: This looks like a different concern. Perhaps it should be in a different commit?

File: pkg/model/openstackmodel/servergroup.go [hunk 2]
```diff
@@ -183,23 +186,12 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
...
 	return nil
 }
 
-func (b *ServerGroupModelBuilder) associateFixedIPToKeypair(fipTask *openstacktasks.Instance) {
-	// Ensure the floating IP is included in the TLS certificate,
-	// if we're not going to use an alias for it
-	fipTask.ForAPIServer = true
-}
-
 func (b *ServerGroupModelBuilder) associateFIPToKeypair(fipTask *openstacktasks.FloatingIP) {
```
Comment (johngmyers) [inline pkg/model/openstackmodel/servergroup.go:195]: I suspect you'll want to remove this as well, but that can be a separate PR.

File: pkg/model/openstackmodel/servergroup.go (hunk 1 above)
Comment (olemarkus) [reply pkg/model/openstackmodel/servergroup.go:166]: Yep. Done.

File: pkg/model/openstackmodel/servergroup.go (hunk 2 above)
Comment (olemarkus) [reply pkg/model/openstackmodel/servergroup.go:195]: Yep. Either in #9554 or a follow-up.

**Comment by olemarkus** [comment]
/hold
Suspect @zetaab wants to review this one as well.

**Comment by zetaab** [comment]
I cannot get this PR working

https://gist.github.com/zetaab/a878b0f4f6f7cd18dc174231d1731eda

**Comment by rifelpet** [comment]
Retrying the MacOS GHA job fixed the failure but there's now a merge conflict

File: pkg/model/openstackmodel/network.go [hunk 3]
```diff
@@ -50,6 +50,10 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	}
 
 	needRouter := true
+	//Do not need router if there is no external network
+	if b.Cluster.Spec.CloudConfig.Openstack.Router.ExternalNetwork == nil {
```
Comment (rifelpet) [inline pkg/model/openstackmodel/network.go:54]: I think you'll want to do additional checking here to avoid a nil pointer dereference

Comment (olemarkus) [reply pkg/model/openstackmodel/network.go]: Yep. Thanks for catching.
